js/private/js_binary.sh.tpl

```diff
@@ -390,6 +390,20 @@ if [ "${JS_BINARY__EXPECTED_EXIT_CODE:-}" ]; then
     fi
 fi
 
+# Use platform line endings in stdout/stderr files
+if [ "$(is_windows)" -eq "1" ]; then
+    SED_WINDOWS_LINE_ENDINGS="s/$/\\r/"
+    TMPFILE=$(mktemp)
+    if [ "${STDOUT_CAPTURE:-}" ]; then
+        cat "$STDOUT_CAPTURE" | sed "$SED_WINDOWS_LINE_ENDINGS" >"$TMPFILE"
```

Comment:
This feels strange vs just printing the right thing in the first place. Are you sure we can't fix at the spot the \n was printed?

Reply:
I attempted to put this sed pipe above in call to node when we write to stdout/err in the first place, but I had difficulty getting the exit code of the node process. Normally you'd use `PIPESTATUS` get the error code of something earlier in the pipe, but the fact that we run the node process in the background with `&` makes that tricker and I couldn't figure it out.

Reply:
that doesn't seem like the spot where the printing happens though. shouldn't the program inside the node process write the correct thing?

Reply:
If the stdout/stderr files are depended upon by non-test targets, then it would break hermeticity to change the line-endings like this, right?

Reply:
I think that would make it fail on linux/osx. If we just want to fix the test, then telling `write_file` that the expected stdout file should have unix line endings would be the most straightforward.

Reply:
The original test in question was `test8`, which is no longer in the diff.

Reply:
let's go with the other solution then, I'd really like to avoid adding this complexity here